### ISSUE TITLE
Series namespaces

### ISSUE THREAD

**Issue by Zamubafoo** [Pewpews/happypanda#28]
Not all works span a single tankobon. For example Shiwasu no Okina's _Shining Musume_ spans 7 volumes/tankobon and several spin-offs/side stories. 

It'd be great if we could have a Series Namespace and have a UI option to group and limit the series to a single image/slot. This way we would see less repetition and more variety.

**Comment by ghost** [comment]
That's certainly true, but don't you want separate tags for each volume? Perhaps not. I can think of a few doujinshi series that have 10+ in their sequence of one continuous story. Would you want something like that combined together? I don't think there are many manga that have more than single volume, but I could be wrong. I'd hazard a guess at 5% have a story that covers more than a single volume.  Not many come mind for me.  If you are going to go along this path, eventually it seems like the logical conclusion would be to essentially make them into folders essential by image. All one series there, another there and so on. I'm probably being silly though. I don't know how you are going through your collection, but personally I'd say the best way is to use tags and reduce the search space. If I wanted to look through everything, I would use Windows Explorer, or similar folder explorer. Then again, I have mine rather organized. Is repetition prevalent for you? If they all have the same name, they should all be grouped together as in, if sorted alphabetically anyway.

**Comment by Zamubafoo** [comment]
It's not about merging volumes into a single entity in the database or the file system, but about merging them in the UI via a specific tag using the proposed series namespace. 
Every constituent entity would remain with it's own tags, but allow for better filtering and organization.

It's caused by the macro-scale problem with **works** being tagged and sorted on a _**release**_ basis, but that's not something this program can address. However, by adding this it helps alleviate the issue.

**Comment by twiddli** [comment]
Hi, very nice suggestion.
I've been thinking of adding a way to distinguish tanks (which can consist of different stories) from a manga/doujin with regular chapters. I've thought of something along the lines of series namespace where you give tanks the tag `tankoubon`, and it'll get an indicator showing it's a tank. When i finally get to implement that, I'll try and see if I can include your suggestion.

**Comment by Zamubafoo** [comment]
I can post an image of a db schema I planned out for sorting my collection where this kinda of addition would be trivial. It might help you out.

**Comment by twiddli** [comment]
That'd be a help, yes.

**Comment by Zamubafoo** [comment]
![entity based tag system](https://cloud.githubusercontent.com/assets/7584122/9587998/9ead77a8-4ff3-11e5-82e1-cb45d757fc47.png)

Just got home from work and altered it after thinking on it a bit. Hope it helps!

The entities table is being used as a global unique id hub to allow for universal tagging of any entity.

**Comment by ghost** [comment]
I apologize. 
It seems I misunderstood.
What you are proposing is very interesting, though I may still misunderstanding some of the specific technical details.
If I am understanding correctly, if all the relevant information is input, then one could something like: 
Easily look if a specific group does a specific author/circle/magazine/content type.
See which magazines the author is serialized and their rate of having new works published
List all magazine chapters/single chapters separately and can look by author and if there a lot, can look if they have a tank out that collects them. If downloading a translated tank, if the tank was tagged or otherwise contained information of all the magazine volumes the chapters were in, that would be an easy reference and wouldn't need to check for dupes.
There is actually a tag for this already on e-hentai/ex, it's called multi-work series. Not sure how well or often it's used though. A lot of other stuff as well, I'm sure. 
Basically, it makes it more encyclopedic and a local reference work, similar to doujinshi.org, only more in-depth, I suppose.
Of course, this all assumes an ideal automated situation or a lot of manual work from what I can tell. 
No idea about its implementation either.

**Comment by Zamubafoo** [comment]
Most of the stuff you are talking about is working with data in a database, which is probably out of the scope of the program. 

The schema diagram is a portion of my personal database schema. Using it as a reference it would allow Pewpews to rework the database into something that would allow for easy namespace addition from his PoV. By allowing for easier namespace additions, it allows for users to organize their collections via whatever manner they choose as well as allowing the tag system to take on the load of generating organizing abilities for multi-volume series, collections/tankobons, magazine-released series, and much more. 

Honestly, you can probably have everything be a tag and move all the table data into the tag space. Then with an 'entity'-'entity tag map' inner join, search all entities in a single query. It would return everything from releases to tankobon to, hypothetically, translation groups, circles, authors, artists, etc.

Though it would require some reworking of the database schema; which probably is not a high priority target for Pewpews as it takes a while to do so specially if he has to implement a updating portion from the old database schema to the new one.
